module/icp/io/skein_mod.c

```diff
@@ -231,6 +231,19 @@ skein_mod_init(void)
 
 int
 skein_mod_fini(void) {
+	int ret;
+
+	if (skein_prov_handle != 0) {
+		if ((ret = crypto_unregister_provider(skein_prov_handle)) !=
+		    CRYPTO_SUCCESS) {
+			cmn_err(CE_WARN,
+			    "skein _fini: crypto_unregister_provider() "
+			    "failed (0x%x)", ret);
+			return (EBUSY);
```

Comment:
Just for future reference in the core zfs.ko code most places where a const errno is returned the SET_ERROR() macro is used to log it.  `return (SET_ERROR(EBUSY));`.  That said, none of the existing icp code does this so let's avoid doing that here.  No change needed.